LoopBackTests/LBUserTests.m

```diff
@@ -53,61 +62,187 @@ @implementation LBUserTests
  */
 + (id)defaultTestSuite {
     SenTestSuite *suite = [SenTestSuite testSuiteWithName:@"TestSuite for LBContainer."];
-    [suite addTest:[self testCaseWithSelector:@selector(testCreate)]];
-    [suite addTest:[self testCaseWithSelector:@selector(testLogin)]];
-    [suite addTest:[self testCaseWithSelector:@selector(testLogout)]];
-    [suite addTest:[self testCaseWithSelector:@selector(testRemove)]];
+    [suite addTest:[self testCaseWithSelector:@selector(testCreateSaveRemove)]];
+    [suite addTest:[self testCaseWithSelector:@selector(testLoginLogout)]];
+    [suite addTest:[self testCaseWithSelector:@selector(testSetsCurrentUserIdOnLogin)]];
+    [suite addTest:[self testCaseWithSelector:@selector(testCurrentUserIdIsStoredInSharedPreferences)]];
+    [suite addTest:[self testCaseWithSelector:@selector(testClearsCurrentUserIdOnLogout)]];
+    [suite addTest:[self testCaseWithSelector:@selector(testGetCachedCurrentUserReturnsNilInitially)]];
+    [suite addTest:[self testCaseWithSelector:@selector(testFindCurrentUserReturnsNilWhenNotLoggedIn)]];
+    [suite addTest:[self testCaseWithSelector:@selector(testFindCurrentUserReturnsCorrectValue)]];
+    [suite addTest:[self testCaseWithSelector:@selector(testGetCachedCurrentUserReturnsValueLoadedByFindCurrentUser)]];
+    [suite addTest:[self testCaseWithSelector:@selector(testGetCachedCurrentUserReturnsValueLoadedByLogin)]];
+    [suite addTest:[self testCaseWithSelector:@selector(testCachedCurrentUserIsClearedOnLogout)]];
     return suite;
 }
 
-
 - (void)setUp {
     [super setUp];
-    
-    LBRESTAdapter *adapter = [LBRESTAdapter adapterWithURL:[NSURL URLWithString:@"http://localhost:3000"]];
-    self.repository = (CustomerRepository*)[adapter repositoryWithClass:[CustomerRepository class]];
+    // forcibly clear the stored user id
+    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+    [defaults removeObjectForKey:DEFAULTS_CURRENT_USER_ID_KEY];
+
+    self.adapter = [LBRESTAdapter adapterWithURL:[NSURL URLWithString:@"http://localhost:3000"]];
+    self.repository = (CustomerRepository*)[self.adapter repositoryWithClass:[CustomerRepository class]];
 }
 
 - (void)tearDown {
     [super tearDown];
 }
 
-- (void)testCreate {
+- (void)testCreateSaveRemove {
     ASYNC_TEST_START
-    Customer __block *user = (Customer*)[self.repository createUserWithEmail:@"testUser@test.com"
-                                                                    password:@"test"];
-    [user saveWithSuccess:^{
+    double uid = [[NSDate date] timeIntervalSince1970];
+    NSString *userEmail = [NSString stringWithFormat:@"%f%@", uid, USER_EMAIL_DOMAIN];
+
+    Customer __block *customer = (Customer*)[self.repository createUserWithEmail:userEmail password:USER_PASSWORD];
```

Comment:
It's better to keep the lines at 80 characters max, that way they fit in the diff windows on GitHub and the reviewer does not have to scroll the text horizontally.